python/ray/workflow/tests/test_inplace_workflows.py

```diff
@@ -63,6 +63,20 @@ def test_inplace_workflows(workflow_start_regular_shared):
     assert exp_remote.step(k, n).run() == k * 2 ** n
 
 
+def test_tail_recursion_optimization(workflow_start_regular_shared):
+    @workflow.step
+    def tail_recursion(n):
+        import inspect
+
+        # check if the stack is growing
+        assert len(inspect.stack(0)) < 20
+        if n <= 0:
+            return "ok"
+        return tail_recursion.options(allow_inplace=True).step(n - 1)
```

Comment:
Is this test really testing the inplace running? I think in place running is to avoid running the function on a remote task right? If so, the stack shouldn't increase I think?🤔

Reply:
Actually running the function on a remote task would not increase the task. Inplace call would increase the stack because of local function calls.